pkg/cloud/cloud.go

```diff
@@ -321,25 +321,25 @@ func (c *cloud) CreateDisk(ctx context.Context, volumeName string, diskOptions *
 
 	// EBS doesn't handle empty outpost arn, so we have to include it only when it's non-empty
 	if len(diskOptions.OutpostArn) > 0 {
-		request.OutpostArn = aws.String(diskOptions.OutpostArn)
+		req.OutpostArn = aws.String(diskOptions.OutpostArn)
 	}
 
 	if len(diskOptions.KmsKeyID) > 0 {
-		request.KmsKeyId = aws.String(diskOptions.KmsKeyID)
-		request.Encrypted = aws.Bool(true)
+		req.KmsKeyId = aws.String(diskOptions.KmsKeyID)
+		req.Encrypted = aws.Bool(true)
 	}
 	if iops > 0 {
-		request.Iops = aws.Int64(iops)
+		req.Iops = aws.Int64(iops)
 	}
 	if throughput > 0 && diskOptions.VolumeType == VolumeTypeGP3 {
-		request.Throughput = aws.Int64(throughput)
+		req.Throughput = aws.Int64(throughput)
 	}
 	snapshotID := diskOptions.SnapshotID
 	if len(snapshotID) > 0 {
-		request.SnapshotId = aws.String(snapshotID)
+		req.SnapshotId = aws.String(snapshotID)
 	}
 
-	response, err := c.ec2.CreateVolumeWithContext(ctx, request)
+	response, err := c.ec2.CreateVolumeWithContext(ctx, req, request.WithAppendUserAgent(userAgent()))
```

Comment:
this shouldn't be necessary, i'm sure there's a way to create the client such that all requests have teh useragent so that you don't have to add the useragent to every single request

Reply:
you can write your own handler and add it tot he chain or leverage the existing handler and add it to the chain https://github.com/aws/aws-sdk-go/blob/v1.38.49/aws/corehandlers/user_agent.go#L22

Reply:
https://github.com/aws/aws-sdk-go/blob/e2d6cb448883e4f4fcc5246650f89bde349041ec/aws/defaults/defaults.go#L77

AddHostExecEnvUserAgentHander is already one of the default handlers